homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,214 @@
+"""Support for Somfy Covers."""
+
+import logging
+from typing import Any, Dict, List, Optional
+
+from pymfy.api.devices.category import Category
+from pymfy.api.devices.thermostat import (
+    DurationType,
+    HvacState,
+    RegulationState,
+    TargetMode,
+    Thermostat,
+)
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_SLEEP,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_BATTERY_LEVEL, ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import SomfyEntity
+from .const import API, COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_CATEGORIES = {Category.HVAC.value}
+
+PRESET_FROST_GUARD = "Frost Guard"
+PRESET_GEOFENCING = "Geofencing"
+PRESET_MANUAL = "Manual"
+
+PRESETS_MAPPING = {
+    TargetMode.AT_HOME: PRESET_HOME,
+    TargetMode.AWAY: PRESET_AWAY,
+    TargetMode.SLEEP: PRESET_SLEEP,
+    TargetMode.MANUAL: PRESET_MANUAL,
+    TargetMode.GEOFENCING: PRESET_GEOFENCING,
+    TargetMode.FROST_PROTECTION: PRESET_FROST_GUARD,
+}
+REVERSE_PRESET_MAPPING = {v: k for k, v in PRESETS_MAPPING.items()}
+
+HVAC_MODES_MAPPING = {HvacState.COOL: HVAC_MODE_COOL, HvacState.HEAT: HVAC_MODE_HEAT}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy cover platform."""
+
+    def get_thermostats():
+        """Retrieve thermostats."""
+        domain_data = hass.data[DOMAIN]
+        coordinator = domain_data[COORDINATOR]
+        api = domain_data[API]
+
+        return [
+            SomfyClimate(coordinator, device_id, api)
+            for device_id, device in coordinator.data.items()
+            if SUPPORTED_CATEGORIES & set(device.categories)
+        ]
+
+    async_add_entities(await hass.async_add_executor_job(get_thermostats))
+
+
+class SomfyClimate(SomfyEntity, ClimateEntity):
+    """Representation of a Somfy thermostat device."""
+
+    def __init__(self, coordinator, device_id, api):
+        """Initialize the Somfy device."""
+        super().__init__(coordinator, device_id, api)
+        self.climate = None
```

Comment:
I believe this is non-public, used only within SomfyClimate, so it can be prefixed with an underscore.

Reply:
You are right, I will do the same for the cover platform.